server/paymentProviders/transferwise/index.ts

```diff
@@ -0,0 +1,75 @@
+import uuidv4 from 'uuid/v4';
+
+import * as transferwise from '../../lib/transferwise';
+import { Quote } from '../../types/transferwise';
+
+async function populateProfileId(connectedAccount): Promise<void> {
+  if (!connectedAccount.data.profile) {
+    const profiles = await transferwise.getProfiles(connectedAccount.token);
+    const profile =
+      profiles.find(p => p.type === connectedAccount.type) || profiles.find(p => p.type === 'business') || profiles[0];
+    if (profile) {
+      connectedAccount.set({ data: { ...connectedAccount.data, ...profile } });
+      await connectedAccount.save();
```

Comment:
Not a requested change: this can also be written with a single `update` line 

```es6
await connectedAccount.update({ data: { ...connectedAccount.data, ...profile } });
```

Reply:
Not really, I'm counting on the fact that this method will mutate `connectedAccount`.
This means `populateProfileId()` is not a pure function, but if that's a problem I can rewrite the logic around this.

Reply:
`update` does mutate the object, as per the docs:

> This is the same as calling set and then calling save but it only saves the exact values passed to it, making it more atomic and safer.